azurerm/internal/services/dns/dns_zone_data_source.go

```diff
@@ -77,16 +77,17 @@ func dataSourceArmDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 			return fmt.Errorf("Error reading DNS Zone %q (Resource Group %q): %+v", name, resourceGroup, err)
 		}
 	} else {
-		rgClient := meta.(*clients.Client).Resource.GroupsClient
-
-		resp, resourceGroup, err = findZone(client, rgClient, ctx, name)
+		var zone *dns.Zone
+		zone, resourceGroup, err = findZone(client, ctx, name)
 		if err != nil {
 			return err
 		}
 
-		if resourceGroup == "" {
+		if zone == nil {
 			return fmt.Errorf("Error: DNS Zone %q was not found", name)
 		}
+
+		resp = *zone
 	}
```

Comment:
@tombuildsstuff thanks for your suggestions. I have updated this function to return a pointer. In order to use the same variable `resp` and `resourcegroup`, I made a little update here, please have a check again.